### PR TITLE
Cleaning zip packages correctly during uninstall/update.

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
Cleaning up zip packages correctly during uninstall and update.

---

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/azure/walinuxagent/1642)

### CODE REVIEW

**Comment by narrieta** [comment]
@vrdmr Travis failed

**Comment by vrdmr** [comment]
@narrieta - Yes. Working on fixing the tests.

File: tests/ga/test_extension.py [hunk 1]
```diff
@@ -435,6 +443,114 @@ def test_ext_handler(self, *args):
...
+        # Test uninstall
+        test_data.goal_state = test_data.goal_state.replace("<Incarnation>1<",
+                                                            "<Incarnation>2<")
+        test_data.ext_conf = test_data.ext_conf.replace("enabled", "uninstall")
+        exthandlers_handler.run()
+        self._assert_no_handler_status(protocol.report_vm_status)
+        self._assert_ext_pkg_file_status(expected_to_be_present=False,
+                                         extension_version=extension_version)
+
+    def test_ext_zip_file_packages_removed_in_updates_and_uninstall_case(self, *args):
```
Comment (pgombar) [inline tests/ga/test_extension.py]: nit: update instead of updates?

File: azurelinuxagent/ga/exthandlers.py [hunk 2]
```diff
@@ -986,16 +987,14 @@ def uninstall(self):
...
-            if os.path.exists(destination):
-                self.pkg_file = destination
-                os.remove(self.pkg_file)
+            zip_filename = os.path.join(conf.get_lib_dir(), self.get_extension_package_zipfile_name())
+            if os.path.exists(zip_filename):
+                os.remove(zip_filename)
+                self.logger.verbose("Deleted the extension zip at path {0}", zip_filename)
 
-            base_dir = self.get_base_dir()
+            base_dir = self.get_base_dir()
```
Comment (pgombar) [inline azurelinuxagent/ga/exthandlers.py]: nit: extra whitespace

**Pull request merged**